### PR TITLE
Adjust post endpoints to reflect changes at user model

### PR DESCRIPTION
**Pull request by Naraujo13**
Changed author name composition with first and last to single name attribute to reflect user model changes

### PR Checklist:

Hey there! It's great to have you here. Here's a quick checklist to help you make a shiny PR:

* [X] Have you checked out the guidelines in our [Contributing](https://github.com/FightPandemics/FightPandemics/blob/master/CONTRIBUTING.md) document?
* [X] Have you looked if there might be other open [Pull Requests](https://github.com/FightPandemics/FightPandemics/pulls?q=is%3Apr+is%3Aopen+sort%3Aupdated-desc) for the same update/change?

### Quick Description of the PR

Change author name composition with first and last to single name attribute to reflect user model changes

### An Overview of the Changes

* Change author name composition
* Fix edge case at `GET /posts/:postId`

### CODE REVIEW

**Comment by joshmorel** [comment]
Hi @Naraujo13 . This looks fine but for my understanding is this just to match current state User model or will we be having single `name` in v2 for both organization and individual user?

**Comment by Naraujo13** [comment]
> Hi @Naraujo13 . This looks fine but for my understanding is this just to match current state User model or will we be having single `name` in v2 for both organization and individual user?

@joshmorel Yes, it is to match current user state. 
Initially, V2 was projected with two separate fields, but recently we started discussing if that was really needed.

**Comment by joshmorel** [comment]
> > Hi @Naraujo13 . This looks fine but for my understanding is this just to match current state User model or will we be having single `name` in v2 for both organization and individual user?
> 
> @joshmorel Yes, it is to match current user state.
> Initially, V2 was projected with two separate fields, but recently we started discussing if that was really needed.

Probably would be simpler overall and less confusing for some non-English users so I support this. The benefits of storing first & last name are minimal I suspect.

File: backend/lib/endpoints/posts.js [hunk 1]
```diff
@@ -242,7 +242,7 @@ async function routes(app) {
         throw app.httpErrors.internalServerError();
       }
 
-      const { comments = [], numComments = 0 } = commentQuery;
+      const { comments = [], numComments = 0 } = commentQuery[0];
```
Comment (joshmorel) [inline backend/lib/endpoints/posts.js:245]: I did approve but don't really understand the reason for this specific change. It's hard for me to confirm if this is right until all the v2 data model changes come in.

Comment (Naraujo13) [reply backend/lib/endpoints/posts.js:245]: The `groupBy` stage at the aggregation pipeline with a `null` group criteria will group all records into only one element (this way it also counts the comments).

So for this query, the results will always be in the first position as a single object, with the array of comments as an attribute, this way we would always be looking at the first element here.

**Comment by Naraujo13** [comment]
> > > Hi @Naraujo13 . This looks fine but for my understanding is this just to match current state User model or will we be having single `name` in v2 for both organization and individual user?
> > 
> > 
> > @joshmorel Yes, it is to match current user state.
> > Initially, V2 was projected with two separate fields, but recently we started discussing if that was really needed.
> 
> Probably would be simpler overall and less confusing for some non-English users so I support this. The benefits of storing first & last name are minimal I suspect.

Currently, the main defender of first + last name is the marketing team who probably wants do differentiate between them to use at campaigns and etc, but I think a single field might be the better option.

**Pull request merged**